src/index.native.tsx

```diff
@@ -1,619 +0,0 @@
-import {
-  Animated,
-  Image,
-  ImageProps,
-  Platform,
-  StyleProp,
-  StyleSheet,
-  UIManager,
-  View,
-  ViewProps,
-  ViewStyle,
-} from 'react-native';
-import { Freeze } from 'react-freeze';
-import { version } from 'react-native/package.json';
-
-import TransitionProgressContext from './TransitionProgressContext';
-import useTransitionProgress from './useTransitionProgress';
-import {
-  StackPresentationTypes,
-  StackAnimationTypes,
-  BlurEffectTypes,
-  ScreenReplaceTypes,
-  ScreenOrientationTypes,
-  HeaderSubviewTypes,
-  ScreenProps,
-  ScreenContainerProps,
-  ScreenStackProps,
-  ScreenStackHeaderConfigProps,
-  SearchBarProps,
-  SearchBarCommands,
-} from './types';
-import {
-  isSearchBarAvailableForCurrentPlatform,
-  isNewBackTitleImplementation,
-  executeNativeBackPress,
-} from './utils';
-
-// web implementation is taken from `index.tsx`
-const isPlatformSupported =
-  Platform.OS === 'ios' ||
-  Platform.OS === 'android' ||
-  Platform.OS === 'windows';
-
-let ENABLE_SCREENS = isPlatformSupported;
-
-function enableScreens(shouldEnableScreens = true): void {
-  ENABLE_SCREENS = isPlatformSupported && shouldEnableScreens;
-  if (ENABLE_SCREENS && !UIManager.getViewManagerConfig('RNSScreen')) {
-    console.error(
-      `Screen native module hasn't been linked. Please check the react-native-screens README for more details`
-    );
-  }
-}
-
-let ENABLE_FREEZE = false;
-
-function enableFreeze(shouldEnableReactFreeze = true): void {
-  const minor = parseInt(version.split('.')[1]); // eg. takes 66 from '0.66.0'
-
-  // react-freeze requires react-native >=0.64, react-native from main is 0.0.0
-  if (!(minor === 0 || minor >= 64) && shouldEnableReactFreeze) {
-    console.warn(
-      'react-freeze library requires at least react-native 0.64. Please upgrade your react-native version in order to use this feature.'
-    );
-  }
-
-  ENABLE_FREEZE = shouldEnableReactFreeze;
-}
-
-// const that tells if the library should use new implementation, will be undefined for older versions
-const shouldUseActivityState = true;
-
-function screensEnabled(): boolean {
-  return ENABLE_SCREENS;
-}
-
-type SearchBarCommandsType = {
-  blur: (
-    viewRef: React.ElementRef<typeof ScreensNativeModules.NativeSearchBar>
-  ) => void;
-  focus: (
-    viewRef: React.ElementRef<typeof ScreensNativeModules.NativeSearchBar>
-  ) => void;
-  clearText: (
-    viewRef: React.ElementRef<typeof ScreensNativeModules.NativeSearchBar>
-  ) => void;
-  toggleCancelButton: (
-    viewRef: React.ElementRef<typeof ScreensNativeModules.NativeSearchBar>,
-    flag: boolean
-  ) => void;
-  setText: (
-    viewRef: React.ElementRef<typeof ScreensNativeModules.NativeSearchBar>,
-    text: string
-  ) => void;
-};
-
-// We initialize these lazily so that importing the module doesn't throw error when not linked
-// This is necessary coz libraries such as React Navigation import the library where it may not be enabled
```

Comment:
How is this handled now? Was the point this comment makes considered?

Reply:
☝🏻 This looks like it might be blocking issue. Are we safe here, also cc @WoLewicki so that you see my concerns, as I'm not fully aware on mechanisms that this comment describes.

Reply:
It was used to lazily load native components from RNScreens, but since we cannot use inline requires right now, and almost all of those components are used from the beginning, I wouldn't care too much about it. Other libs did not have such mechanism iirc.